src/Vector3.h

```diff
@@ -393,11 +393,9 @@ class VectorHasher
 	{
 		// Guaranteed to have no hash collisions for any 128x128x128 area
 		size_t Hash = 0;
-		Hash ^= static_cast<size_t>(a_Vector.x);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.y);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.z);
+		Hash ^= a_Vector.x + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.y + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.z + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
```

Comment:
@worktycho The old hash is more than enough for its use case, we never put vectors in a hashtable other than in the pathfinder, and the 128x128x128 guarantee is enough for that. What exactly is the purpose of this proposed change or (@tigerw 's change), and is it related to the PR in any way?